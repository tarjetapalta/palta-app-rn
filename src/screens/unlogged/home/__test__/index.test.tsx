import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import WithTheme from '@theme/context';
import { Home } from '../index';

jest.mock('react-native-gesture-handler', () => {});

test('renders home correctly', () => {
  const app = renderer
    .create(
      <WithTheme>
        <Home />
      </WithTheme>,
    )
    .toJSON();
  expect(app).toMatchSnapshot();
});
