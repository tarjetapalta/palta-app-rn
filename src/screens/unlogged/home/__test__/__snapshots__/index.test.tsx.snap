// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home correctly 1`] = `
Array [
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/background-02.png",
      }
    }
    style={
      Array [
        Object {
          "height": "100%",
          "position": "absolute",
          "width": "100%",
        },
      ]
    }
  />,
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
    transparent={true}
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "space-between",
          },
        ]
      }
    >
      <View
        content="left"
        style={
          Array [
            Object {
              "alignItems": "flex-start",
              "paddingBottom": 0,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 48,
            },
          ]
        }
        type="header"
      >
        <View
          style={
            Array [
              Object {
                "marginLeft": -15,
              },
            ]
          }
        >
          <View
            height={110}
            style={
              Array [
                Object {
                  "height": 110,
                  "width": 110,
                },
              ]
            }
            width={110}
          >
            <Image
              source={
                Object {
                  "testUri": "../../../assets/images/isologo-512-04.png",
                }
              }
              style={
                Object {
                  "flex": 1,
                  "height": undefined,
                  "resizeMode": "contain",
                  "width": undefined,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "height": 24,
              },
            ]
          }
        />
        <Text
          color="white"
          numberOfLines={0}
          size="xl"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "BwModelica-Bold",
                "fontSize": 24,
                "opacity": 0.87,
                "textAlign": "center",
              },
            ]
          }
          testID="H2"
          weight="bold"
        >
          Billetera virtual.
        </Text>
        <Text
          color="white"
          numberOfLines={0}
          size="xl"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "BwModelica-Bold",
                "fontSize": 24,
                "opacity": 0.87,
                "textAlign": "center",
              },
            ]
          }
          testID="H2"
          weight="bold"
        >
          Dinero real
        </Text>
      </View>
      <View
        content="center"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 0,
              "paddingLeft": 48,
              "paddingRight": 48,
              "paddingTop": 0,
            },
          ]
        }
      />
      <View
        content="center"
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
              "paddingBottom": 48,
              "paddingLeft": 8,
              "paddingRight": 8,
              "paddingTop": 0,
            },
          ]
        }
        type="footer"
      >
        <Text
          color="white"
          emphasis="pure"
          numberOfLines={0}
          size="s"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "BwModelica-Regular",
                "fontSize": 14,
                "opacity": 1,
                "textAlign": "center",
              },
            ]
          }
          testID="H5"
        >
          No tengo cuenta
        </Text>
        <View
          size="s"
          style={
            Array [
              Object {
                "height": 16,
              },
            ]
          }
        />
        <View
          accessibilityLabel="Registrarme"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#333333",
                  "borderRadius": 2,
                  "borderTopColor": "#272727",
                  "borderTopWidth": 4,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "paddingBottom": 2,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "BwModelica-Bold",
                    "fontSize": 14,
                    "letterSpacing": 0.21,
                    "opacity": 0.87,
                    "textAlign": "center",
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Registrarme
            </Text>
          </View>
        </View>
        <View
          size="s"
          style={
            Array [
              Object {
                "height": 16,
              },
            ]
          }
        />
        <Text
          color="white"
          numberOfLines={0}
          size="s"
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontFamily": "BwModelica-Regular",
                "fontSize": 14,
                "opacity": 0.87,
                "textAlign": "center",
              },
            ]
          }
          testID="H5"
        >
          Ya tengo cuenta
        </Text>
        <View
          size="xs"
          style={
            Array [
              Object {
                "height": 8,
              },
            ]
          }
        />
        <View
          accessibilityLabel="Iniciar sesión"
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderRadius": 2,
                  "borderTopColor": "transparent",
                  "borderTopWidth": 4,
                  "flexDirection": "row",
                  "height": 45,
                  "justifyContent": "center",
                  "paddingBottom": 2,
                },
              ]
            }
          >
            <Text
              color="primary"
              style={
                Array [
                  Object {
                    "color": "#91b086",
                    "fontFamily": "BwModelica-Bold",
                    "fontSize": 14,
                    "letterSpacing": 0.21,
                    "opacity": 0.87,
                    "textAlign": "center",
                    "textTransform": "uppercase",
                  },
                ]
              }
            >
              Iniciar sesión
            </Text>
          </View>
        </View>
      </View>
    </RCTSafeAreaView>
  </View>,
]
`;
